apps/web/src/app/[locale]/tracks/_components/track-challenge-card.tsx

```diff
@@ -47,17 +47,22 @@ export function TrackChallenge({ challenge, className, mock }: TrackChallengePro
         )}
       >
         <div className="relative flex items-center gap-3 text-xs sm:text-base">
-          {mock == true && (
-            <>
-              <input
-                className="peer hidden appearance-none"
-                type="checkbox"
-                id={challenge.id.toString()}
-              />
-              <div className="h-5 w-5 rounded-full border border-black/70 bg-black/10 duration-75 peer-checked:border-transparent peer-checked:bg-green-600/80 dark:border-white/50 dark:bg-white/10 peer-checked:dark:bg-green-300/80" />
-              <Check className="absolute left-1 my-auto h-3 w-3 scale-0 stroke-[4] text-white duration-300 peer-checked:scale-100 dark:text-black" />
-            </>
-          )}
+          {(() => {
```

Comment:
That looks like a "hacky fix". We should probably extract everything to another component, so that Million can properly optimize it.

Reply:
Hey I have made the requested changes via making the return deterministic , according to https://million.dev/docs/rules

<img width="1005" alt="Screenshot 2023-09-08 at 6 48 06 PM" src="https://github.com/typehero/typehero/assets/86520455/1697e32b-add0-428b-b597-ccc0870540a1">